internal/providers/azure/azure.go

```diff
@@ -97,19 +108,17 @@ func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config,
 		)
 	}()
 
+	logger.Debug("checking for config drive")
+	if _, err := os.Stat(filepath.Join(mnt, "ovf-env.xml")); err != nil {
+		return nil, fmt.Errorf("failed to identify config drive: %v", err)
```

Comment:
The message is literally true but doesn't really convey what's going on.  How about something like:

```suggestion
		return nil, fmt.Errorf("device %q does not appear to be a config drive: %v", devicePath, err)
```